lib/src/main/java/graphql/nadel/engine/NadelDeferSupport.kt

```diff
@@ -0,0 +1,167 @@
+package graphql.nadel.engine
+
+import graphql.incremental.DelayedIncrementalPartialResult
+import graphql.nadel.engine.NadelDeferSupport.OutstandingJobCounter.OutstandingJobHandle
+import graphql.nadel.engine.util.copy
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.launch
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+
+class NadelDeferSupport internal constructor(
```

Comment:
This class will also be used for `@stream` support. We should name all constructs shared by `@stream` and `@defer` with a common name or prefix. Maybe `Delayed` or `Incremental`?

Reply:
Yep I've renamed it, thanks.